File: elixir/bob/lib/bob.ex
defmodule Bob do

  def hey(input) do
    input = String.trim input
    cond do
      silent?(input) -> "Fine. Be that way!"
      shouting?(input) && question?(input) && letters?(input) -> "Calm down, I know what I'm doing!"
      question?(input) -> "Sure."
      shouting?(input) && letters?(input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end

defp silent?(input), do: input === ""
defp letters?(input), do: String.upcase(input) != String.downcase(input)
defp shouting?(input), do: String.upcase(input) === input
defp question?(input), do: String.ends_with?(input, "?")

end

File: javascript/bob/bob.js
export const hey = (message) => {
  message = message.trim();
  return ANSWERS
    .find(x => x.test(message))
    .answer;
};

const silent = input => input === "";
const letters = input => input.toUpperCase() !== input.toLowerCase(); 
const shouting = input => input.toUpperCase() === input; 
const question = input => input[input.length - 1] === '?'; 

const ANSWERS = [
  { answer: "Fine. Be that way!", test: silent },
  { answer: "Calm down, I know what I'm doing!", test: message => shouting(message) && question(message) && letters(message) },
  { answer: "Sure.", test: question },
  { answer: "Whoa, chill out!", test: message => shouting(message) && letters(message) },
  { answer: "Whatever.", test: () => true }
];